src/tree/namespace.rs

```diff
@@ -16,6 +17,21 @@ pub struct Namespace {
   pub(crate) ns_ptr: xmlNsPtr,
 }
 
+impl PartialEq for Namespace {
+  fn eq(&self, other: &Self) -> bool {
+    self.get_prefix() == other.get_prefix() && self.get_href() == other.get_href()
+  }
+}
```

Comment:
This is a little tricky, since it can be surprising depending on what expectation someone approaches the `Namespace` struct.

Internally, Namespace stores a libxml2 `xmlNsPtr`. So one could have expected pointer equality. Alternatively, one could have expected the equality to be in the XML sense - if the URL is identical, then the namespaces are equal, even if the prefixes are different.

So introducing the "deepest" kind of equality here (same prefix and same namespace) isn't immediately expected behavior. Additionally, it is the slowest variant performance-wise, and uses dedicated allocations.

Hm... I am not opposed to merging it as introduced here, but I wonder if we'll get requests to modify it down the road. For now a clarifying comment on the Namespace struct could help.